docs/ref/models/fields.txt

```diff
@@ -115,9 +115,41 @@ human-readable name. For example::
         ("GR", "Graduate"),
     ]
 
+``choices`` can also be defined as a callable that expects no arguments and
+returns any of the formats described above. For example::
+
+    import functools
+
+
+    @functools.cache
+    def fib(n):
+        if n < 2:
+            return n
+        return fib(n - 1) + fib(n - 2)
```

Comment:
Applying suggestion (though I think @felixxm wanted to propose a different example).

I chose Fibonacci because it was a nice example to make good use of caching, but now that the caching has "losr focus", perhaps I should propose an example that returns a setting or a similar "static-ish" list?